packages/@aws-cdk/aws-sagemaker/test/model-data.test.ts

```diff
@@ -0,0 +1,34 @@
+import * as path from 'path';
+import * as sagemaker from '../lib';
+
+describe('When creating model data from a local asset', () => {
+  test('by supplying a directory, an exception is thrown', () => {
+    // GIVEN
+
+    // WHEN
+    const when = () => sagemaker.ModelData.fromAsset(path.join(__dirname, 'test-artifacts'));
+
+    // THEN
+    expect(when).toThrow(/Asset must be a gzipped tar file with extension .tar.gz/);
+  });
+
+  test('by supplying a zip file, an exception is thrown', () => {
+    // GIVEN
+
+    // WHEN
+    const when = () => sagemaker.ModelData.fromAsset(path.join(__dirname, 'test-artifacts', 'invalid-artifact.zip'));
+
+    // THEN
+    expect(when).toThrow(/Asset must be a gzipped tar file with extension .tar.gz/);
+  });
+
+  test('by supplying a file with an unsupported extension, an exception is thrown', () => {
+    // GIVEN
```

Comment:
super nit: if there's nothing given, we might as well omit this and save 2 lines :)